File: Java/CodingChallenges/LeetCode/two-sum/Attempt02.java
public class Attempt02 {
    public static void main(String[] args) throws Exception {

        // Example 1:
        // int[] nums = { 2, 7, 11, 15 };
        // int target = 9;
        // Output: [0,1]
        // Output: Because nums[0] + nums[1] == 9, we return [0, 1].

        // Example 2:
        int[] nums = { 3, 2, 4 };
        int target = 6;
        // Output: [1,2]

        twoSum(nums, target);
    }

    public static int[] twoSum(int[] nums, int target) {
        // for (int i = 0; i < 10; i++) {
        // if (i == 4) {
        // break;
        // }
        // System.out.println(i);
        // }

        for (int i = 0; i < nums.length; i++) {
            for (int j = 1; j < nums.length; j++) {
                if (nums[i] + nums[j] == target) {
                    System.out.println("yippie");
                    System.out.println("i: " + i);
                    System.out.println("j: " + j);
                }
            }
        }

        // for (int i : nums) {
        // System.out.println("i: "+i);
        // while (j < nums.length) {
        // System.out.println("j: "+ j);
        // System.out.println("i+j: "+ i + j);
        // if (i + nums[j] != target) {
        // j++;
        // // System.out.println("nope");
        // // System.out.println("j: "+ j);
        // // System.out.println("i: "+i);
        // // break;
        // } else {
        // // System.out.println("else");
        // System.out.println("yippie");
        // }
        // }
        // }

        return nums;
    }
}

File: Languages/JavaScript/CodingChallenges/CodeWars/TwoSum/attempt02.js
// attempt01.js

// Brute force solution

// // Example 1:
// // Input: nums = [2,7,11,15], target = 9
// // Output: [0,1]
// // Explanation: Because nums[0] + nums[1] == 9, we return [0, 1].
const numbers = [2, 7, 11, 15];
const target = 9;
// want [0,1]

// // Example 2:
// // Input: nums = [3,2,4], target = 6
// // Output: [1,2]
// const numbers = [3, 2, 4];
// const target = 6;
// // want [1,2]

// // Example 3:
// // Input: nums = [3,3], target = 6
// // Output: [0,1]
// const numbers = [3, 3];
// const target = 6;
// // want [0,1]

// // Example 4:
// // Input: nums = [1, 2, 3], target = 4
// const numbers = [1, 2, 3];
// const target = 4;
// // want [0,2]

var twoSum = function (numbers, target) {
  //   Check:
  //   For a given input array this algorithm does the following steps:
  // Create a hashmap which accepts integer datatype as key and value.
  const map1 = new Map();
  // Iterate through each element in the given array starting from the first element.
  for (let index = 0; index < numbers.length; index++) {
    // In each iteration check if required number (required  number = target sum - current number) is present in the hashmap.
    // console.log("numbers[index]:", numbers[index]);
    // console.log("target - numbers[index]:", target - numbers[index]);
    // console.log(
    //   "map1.get(target - numbers[index]):",
    //   map1.get(target - numbers[index]),
    // );
    if (map1.has(target - numbers[index])) {
      // console.log([map1.get(target - numbers[index]), numbers[index]]);
      // console.log("map1", map1);
      // If present, return {required number index, current number index} as  result.
      return [map1.get(target - numbers[index]), index];
    } else {
      // Otherwise add the current iteration number as key and its index as value to the hashmap. Repeat this  until you find the result.
      map1.set(numbers[index], index);
      // console.log("map1:", map1);
    }
  }
};
console.log(twoSum(numbers, target));
